recipes/tinyalsa/all/patches/tinyalsa-2.0.0-strncpy-warning.patch

```diff
@@ -0,0 +1,39 @@
+diff --git a/src/mixer_plugin.c b/src/mixer_plugin.c
+index 34117a9..b7fcc0b 100644
+--- a/src/mixer_plugin.c
++++ b/src/mixer_plugin.c
+@@ -82,7 +82,7 @@ static int mixer_plug_get_elem_id(struct mixer_plug_data *plug_data,
+     id->iface = ctl->iface;
+ 
+     strncpy((char *)id->name, (char *)ctl->name,
+-            sizeof(id->name));
++            sizeof(id->name) - 1);
```

Comment:
I think you need to write a terminating '\0` to the last element of buffer.

For example, the following source:
```c
#include <string.h>
#include <stdio.h>
static void print_buffer(const char *s, int nb) {
    for(int i=0; i < nb; i += 1) {
        printf("%c(%d) ", s[i], s[i]);
    }
    printf("\n");
}
int main() {
    const char *input = "hello";
    char buffer5[5];
    char buffer6[6];
    strncpy(buffer5, input, 5);
    print_buffer(buffer5, 5);
    strncpy(buffer6, input, 6);
    print_buffer(buffer6, 6);
}
```
produces the following output:
```
h(104) e(101) l(108) l(108) o(111) 
h(104) e(101) l(108) l(108) o(111) (0) 
```
There is no zero terminator when the output buffer is just big enough.

Reply:
Oh, I missed it. 
You are right.
I will make another PR to tinyalsa, then come back this PR.

Reply:
Just one of the many paper cuts why bugs are so prevalent in c sources :cry:

Reply:
Fixed.
https://github.com/tinyalsa/tinyalsa/pull/221